tests/compiletest.rs

```diff
@@ -38,6 +38,9 @@ fn run_tests(mode: Mode, path: &str, target: Option<String>) {
         flags.push(target.clone());
     }
 
+    // add flags for testing the external C functionality
+    flags.push("-Zmiri-external_c_so_file=tests/external_C/libtestlib.so".to_string());
```

Comment:
I think we should add a test to see what happens if the flag is used with missing/wrong files, and if the diagnostics are reasonable, just stick with that.

Which reminds me: this doesn't need to go here. You can add it to just the tests that actually use it by adding a `//@compile-flags: -Zmiri....` comment to them. This will also make filtering the tests for supported platforms easier, as you can now use `//@only-linux` or sth to make sure the tests are only run there